doc/source/development/building_from_source.rst

```diff
@@ -90,6 +84,41 @@ for the shared lib, *e.g.* ``set (GDAL_LIB_OUTPUT_NAME gdal_x64 CACHE STRING ""
 
 Refer to :ref:`using_gdal_in_cmake` for how to use GDAL in a CMake project.
 
+Building on Windows
++++++++++++++++++++
+
+On Windows, one may need to specify generator:
+
+.. code-block:: bash
+
+    cmake -G "Visual Studio 15 2017" ..
+
+
+Building on MacOS
++++++++++++++++++
+
+On MacOS, there are a couple of libraries that do not function properly when the GDAL build requirements are installed using Homebrew.
+
+The `Apache Arrow <https://arrow.apache.org/docs/index.html>`_ library included in the current distribution of CMake is broken, and causes a detection issue. In order to build GDAL successfuly, you must first remove it:
+
+.. code-block:: bash
+
+    rm -rf /usr/local/lib/cmake/Arrow
+
+And then set the appropriate options to exclude Arrow from the build:
+
+.. code-block:: bash
+
+    cmake -DGDAL_USE_ARROW=OFF -DGDAL_USE_PARQUET=OFF ..
```

Comment:
Instead of removing arrow cmake config files, it should be enough to use `-DCMAKE_DISABLE_FIND_PACKAGE_Arrow=ON`.
Related analysis: https://github.com/microsoft/vcpkg/issues/33193

OTOH arrow in vcpkg should work also for macOS.

Reply:
I can confirm this works - after restoring my Arrow config files (and confirming I once again got the original find_package error without the flag), I can successfully build while the Arrow config is present if that flag is included. Thanks for the suggestion, it's much cleaner.

I've updated this section of the docs accordingly.